avogadro/qtplugins/noncovalent/noncovalent.cpp

```diff
@@ -116,15 +125,90 @@ static bool checkHoleVector(
       float oppositeAngle = M_PI - computeAngle(
         in, npos - pos
       );
-      return oppositeAngle < angleTolerance;
+      return oppositeAngle <= angleTolerance;
     }
   );
 }
 
 static bool checkPairVector(
     const Molecule &molecule, Index n, Vector3 in, float angleTolerance
 ) {
-  return true;
+  AtomHybridization hybridization = AtomUtilities::perceiveHybridization(molecule.atom(n));
+  Array<const Bond *> bonds = molecule.bonds(n);
+  size_t bondCount = bonds.size();
+  std::vector<Vector3> bondVectors(bondCount);
+  Vector3 pos = molecule.atomPosition3d(n);
+  std::transform(bonds.begin(), bonds.end(), bondVectors.begin(), [molecule, n, pos](const Bond *b) {
+    return molecule.atomPosition3d(b->getOtherAtom(n).index());
+  });
+  float pairAngle;
```

Comment:
This needs more commenting. What I've learned with similar code is that people (including myself) find it hard to understand why particular angles or math are being done this way.

Reply:
It's also probably a candidate to go into `AtomUtilities` as well.

Reply:
I'm sorry, there was a mistake in this part, I forgot to subtract the central atom's position when rewriting it to use `std::transform`. It's actually just for computing regular bond vectors.